itest/rust/src/object_tests/dyn_gd_test.rs

```diff
@@ -76,16 +95,23 @@ fn dyn_gd_upcast() {
 
     // Ensure applied to the object polymorphically. Concrete object can access bind(), no dyn_bind().
     assert_eq!(original_copy.bind().get_hitpoints(), 75);
+    assert_eq!(original_copy.bind().get_report(), 42.0);
 
     // Check also another angle (via Object). Here dyn_bind().
     assert_eq!(object.dyn_bind().get_hitpoints(), 75);
 
+    let object = concrete_with_associated_type.clone().upcast::<Object>();
+    assert_eq!(object.dyn_bind().get_report(), 42.0);
+
+    // via DynGd with associated type
+    assert_eq!(concrete_with_associated_type.dyn_bind().get_report(), 42.0);
+
     node.free();
 }
 
 #[itest]
 fn dyn_gd_downcast() {
-    let original = Gd::from_object(RefcHealth { hp: 20 }).into_dyn();
+    let original: DynGd<RefcHealth, dyn Health> = Gd::from_object(RefcHealth { hp: 20 }).into_dyn();
```

Comment:
All these tests also checked type inference: when there's only one candidate of a trait, the compiler is smart enough to figure it out. Explicitly specifying types defeats this. 

Would this still be necessary with my `IdProvider` proposal below? Would be cool if you could revert changes to existing tests regarding type inference; these should keep running as-is.